From 73e0e67c8dca2a6fac4d170c582bb4552087ad46 Mon Sep 17 00:00:00 2001
From: Sujeev Dias <sdias@codeaurora.org>
Date: Wed, 7 Feb 2018 14:04:35 -0800
Subject: [PATCH 1/3] mhi_bus: core: Add support for MHI host interface

MHI Host Interface is a communication protocol to be used by the host
to control and communcate with modem over a high speed peripheral bus.
This module will allow host to communicate with external devices that
support MHI protocol.

CRs-Fixed: 2204910
Change-Id: I6057f9167ebbb5d4dc2ebb91e46ede6bc7054325
Signed-off-by: Sujeev Dias <sdias@codeaurora.org>
---
 include/linux/mod_devicetable.h | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/include/linux/mod_devicetable.h b/include/linux/mod_devicetable.h
index 1c2e8d6..40824a4 100644
--- a/include/linux/mod_devicetable.h
+++ b/include/linux/mod_devicetable.h
@@ -687,5 +687,17 @@ struct fsl_mc_device_id {
 	const char obj_type[16];
 };
 
+#define MHI_NAME_SIZE 32
+
+/**
+ * struct mhi_device_id - MHI device identification
+ * @chan: MHI channel name
+ * @driver_data: driver data;
+ */
+
+struct mhi_device_id {
+	const char chan[MHI_NAME_SIZE];
+	kernel_ulong_t driver_data;
+};
 
 #endif /* LINUX_MOD_DEVICETABLE_H */
-- 
1.9.1

